backend/src/main/java/gov/cdc/usds/simplereport/service/ApiUserService.java

```diff
@@ -213,17 +213,43 @@ private ApiUser getPatientApiUser() {
     }
   }
 
+  private static final String NOREPLY = "-noreply@simplereport.gov";
+  private static final String PATIENT_REGISTRATION_EMAIL = "patient-registration" + NOREPLY;
+
   private String getPatientIdEmail(Person patient) {
```

Comment:
for my own edification -- will line 194 of this file ever return a found apiUser if the patient has self-registered? given that the self-registration apiUser's email does not contain a UUID but the patientIdEmail always does contain a UUID.

Reply:
Great question! **Yes, it will.** The magic self-registration ApiUser exists only to fill the `created_by` field of Persons that self-register. PatientApiUsers, of the `UUID-noreply@simplereport.gov` variety, are created to allow that Person to be make requests without exploding the audit machinery, and they're created on-demand for that Person.

So if a patient self-registers, they'll have `created_by: self-registration-noreply@simplereport.gov`. If they are then sent their Time of Test questions via SMS, and they log in by authenticating their DOB, the `UUID-noreply@simplereport.gov` ApiUser will be created. If that user updates their demographic details, they'll end up with an `updated_by: UUID-noreply@simplereport.gov`

Does that answer your question? I'm worried that I've just muddied the waters 😂

Reply:
yup! thank you!